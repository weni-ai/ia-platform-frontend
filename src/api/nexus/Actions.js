import request from '@/api/nexusaiRequest';

const groups = {
  'Interações gerais': 'interactions',
  'Compras de Produtos': 'shopping',
  'Assistência e Suporte': 'support',
};

export const Actions = {
  types: {
    async list({ projectUuid, language }) {
      const { data } = await request.$http.get(
        `api/${projectUuid}/template-action/`,
        {
          params: {
            language: language === 'en-us' ? 'en' : language,
          },
          hideGenericErrorAlert: true,
        },
      );

      return data
        .map(
          ({ uuid, name, display_prompt, action_type, group, language }) => ({
            uuid,
            name,
            prompt: display_prompt,
            type: action_type,
            group: Object.values(groups).includes(group) ? group : 'custom',
          }),
        )
        .filter(({ group }) => group);
    },
  },

  async create({ projectUuid, flowUuid, templateUuid, name, prompt, type }) {
    const { data } = await request.$http.post(`api/${projectUuid}/flows/`, {
      uuid: flowUuid,
      action_template_uuid: templateUuid,
      name,
      prompt,
      action_type: type,
      fallback: false,
    });

    return {
      uuid: data.uuid,
      name: data.name,
      prompt: data.prompt,
      type: data.action_type,
      group:
        templateUuid && Object.values(groups).includes(data.group)
          ? data.group
          : 'custom',
    };
  },

  async edit({ projectUuid, actionUuid, name, prompt }) {
    const { data } = await request.$http.patch(
      `api/${projectUuid}/flows/${actionUuid}/`,
      {
        name,
        prompt,
      },
    );

    return {
      uuid: data.uuid,
      name: data.name,
      prompt: data.prompt,
    };
  },

  delete({ projectUuid, actionUuid }) {
    return request.$http.delete(`api/${projectUuid}/flows/${actionUuid}/`);
  },

  async list({ projectUuid }) {
    const { data } = await request.$http.get(`api/${projectUuid}/flows/`);

    return data.map(
      ({
        uuid,
        name,
        prompt,
        action_type,
        editable,
        content_base,
        fallback,
        group,
      }) => ({
        uuid,
        name,
        prompt,
        type: action_type,
        editable,
        group: Object.values(groups).includes(group) ? group : 'custom',
      }),
    );
  },

  generatedNames: {
    generate({ projectUuid, chatbot_goal, context }) {
      return request.$http.post(`api/${projectUuid}/generate-action-name/`, {
        chatbot_goal,
        context,
      });
    },
  },

  flows: {
    list({ next, projectUuid, name }) {
      if (next) {
        return request.$http.get(`api/${projectUuid}/search-flows/${next}`);
      }

      return request.$http.get(`api/${projectUuid}/search-flows/`, {
        params: {
          name: name ? name : undefined,
        },
      });
    },
  },
};
