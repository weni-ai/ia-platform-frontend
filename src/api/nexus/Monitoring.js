import request from '@/api/nexusaiRequest';
import { cleanParams } from '@/utils/http';

export const Monitoring = {
  messages: {
    async list({
      projectUuid,
      page,
      pageInterval,
      tag,
      text,
      started_day,
      ended_day,
    }) {
      const params = cleanParams({
        tag,
        text,
        started_day,
        ended_day,
        page_size: pageInterval,
        page,
      });
      const {
        data: { results, count },
      } = await request.$http.get(`api/${projectUuid}/message_history/`, {
        params,
      });

      return {
        count,
        data: results.map(
          ({ id, created_at, message_text, tag, classification }) => ({
            created_at,
            id,
            text: message_text,
            tag,
            action_name: classification,
          }),
        ),
      };
    },

    async history({ projectUuid, id }) {
      // const {
      //   data: { results },
      // } = await request.$http.get(
      //   `api/${projectUuid}/message_history/?uuid=${id}`,
      // );

      return {
        uuid: 'uuid',
        contact_urn: 'whatsapp:999999999',
        text: 'como envio meu curr√≠culo??',
        llm_response:
          'Desculpe, mas essa pergunta est√° fora do meu escopo de conhecimento. Posso te ajudar com outras d√∫vidas relacionadas a lhamas? ü¶ô',
        llm_response_status: 'failed',
      };
    },

    async performance({ projectUuid, started_day, ended_day }) {
      const params = cleanParams({
        started_day,
        ended_day,
      });
      const { data } = await request.$http.get(
        `api/${projectUuid}/tags-analytics/`,
        {
          params,
        },
      );

      return data;
    },
  },
};
