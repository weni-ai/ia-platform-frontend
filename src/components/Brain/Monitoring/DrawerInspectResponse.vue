<template>
  <UnnnicDrawer
    :modelValue="modelValue"
    :title="$t('router.monitoring.inspect_response.title')"
    :description="
      $t(
        'router.monitoring.inspect_response.intelligent_agent_response_details',
      )
    "
    @close="close"
  >
    <template #content>
      <section class="inspect-response">
        <section class="inspect-response__contact">
          <UnnnicIntelligenceText
            color="neutral-cloudy"
            family="secondary"
            size="body-gt"
            tag="p"
          >
            {{ $t('router.monitoring.inspect_response.contact_message') }}:
          </UnnnicIntelligenceText>
          <p class="contact__text">“{{ inspectionData.text }}”</p>
        </section>
        <section class="inspect-response__agent">
          <UnnnicIntelligenceText
            color="neutral-cloudy"
            family="secondary"
            size="body-gt"
            tag="p"
          >
            {{ $t('router.monitoring.inspect_response.agent_response') }}:
          </UnnnicIntelligenceText>
          <p
            v-if="inspectionData.llm.status !== 'success'"
            class="agent__text"
          >
            “{{ inspectionData.llm.response }}”
          </p>
          <section v-else>
            <template
              v-for="fragment of groundednessFragments"
              :key="fragment.sentence"
            >
              <p
                :class="[
                  'agent__text',
                  'agent__text--underlined',
                  `agent__text--underlined-${getReliabilityLevel(fragment)}`,
                ]"
              >
                {{ fragment.sentence }}
              </p>
            </template>
          </section>
        </section>
      </section>
    </template>
  </UnnnicDrawer>
</template>

<script setup>
import { computed } from 'vue';

const props = defineProps({
  modelValue: {
    type: Boolean,
    default: false,
  },

  inspectionData: {
    type: Object,
    required: true,
  },
});

const emit = defineEmits(['update:model-value']);

const groundednessFragments = computed(() => {
  const data = props.inspectionData.groundedness.map((fragment) => ({
    ...fragment, // Required to create copies of the fragments and not update the prop
  }));

  if (data.length > 0) {
    const firstFragment = data[0];
    const lastFragment = data.at(-1);

    firstFragment.sentence = `“${firstFragment.sentence}`;
    lastFragment.sentence = `${lastFragment.sentence}”`;
  }

  return data;
});

function close() {
  emit('update:model-value', false);
}

function getReliabilityLevel(item) {
  const { score } = item;

  if (score === 0) return 'unreliable';
  if (score <= 33) return 'slightly-reliable';
  if (score <= 66) return 'moderately-reliable';
  if (score <= 100) return 'highly-reliable';
  return 'unreliable';
}
</script>

<style lang="scss" scoped>
.inspect-response {
  display: flex;
  flex-direction: column;
  gap: $unnnic-spacing-md;

  &__contact,
  &__agent {
    display: grid;
    gap: $unnnic-spacing-nano;

    .contact__text,
    .agent__text {
      color: $unnnic-color-neutral-darkest;
      font-size: $unnnic-font-size-body-lg;
    }
  }

  &__agent {
    .agent__text {
      display: inline;

      line-height: $unnnic-font-size-body-lg + $unnnic-line-height-md;

      &--underlined {
        margin-right: $unnnic-spacing-nano;

        text-decoration: underline;
        text-decoration-color: $unnnic-color-neutral-cleanest;
        text-decoration-thickness: 3px;
        text-underline-offset: 4px;

        &-slightly-reliable {
          text-decoration-color: $unnnic-color-weni-300;
        }
        &-moderately-reliable {
          text-decoration-color: $unnnic-color-weni-600;
        }
        &-highly-reliable {
          text-decoration-color: $unnnic-color-weni-800;
        }
      }
    }
  }
}
</style>
