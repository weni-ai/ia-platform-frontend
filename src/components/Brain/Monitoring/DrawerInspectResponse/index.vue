<template>
  <UnnnicDrawer
    data-testid="drawer"
    :modelValue="modelValue"
    :title="$t('router.monitoring.inspect_response.title')"
    :description="
      $t(
        'router.monitoring.inspect_response.intelligent_agent_response_details',
      )
    "
    @close="close"
  >
    <template #content>
      <section class="inspect-response">
        <ContactMessage :text="inspectionData.text" />
        <ActionDetails
          v-if="llmStatus === 'action'"
          :action="inspectionData.action"
        />
        <AgentResponse
          v-if="llmStatus !== 'action'"
          :status="llmStatus"
          :response="inspectionData.llm.response"
          :inspectionData="inspectionData"
        />
        <ResponseEvaluator
          v-if="llmStatus !== 'failed'"
          :isApproved="inspectionData.is_approved"
        />
      </section>
    </template>
  </UnnnicDrawer>
</template>

<script setup>
import { computed } from 'vue';

import ContactMessage from './ContactMessage.vue';
import ActionDetails from './ActionDetails.vue';
import AgentResponse from './AgentResponse.vue';
import ResponseEvaluator from './ResponseEvaluator.vue';

const props = defineProps({
  modelValue: {
    type: Boolean,
    default: false,
  },
  inspectionData: {
    type: Object,
    required: true,
  },
});

const emit = defineEmits(['update:model-value']);

const llmStatus = computed(() => props.inspectionData.llm.status);

function close() {
  emit('update:model-value', false);
}
</script>

<style lang="scss" scoped>
.inspect-response {
  display: flex;
  flex-direction: column;
  gap: $unnnic-spacing-md;

  &__contact,
  &__agent {
    display: grid;
    gap: $unnnic-spacing-nano;

    .contact__text,
    .agent__text {
      color: $unnnic-color-neutral-darkest;
      font-size: $unnnic-font-size-body-lg;
    }
  }

  &__agent .agent__text {
    display: inline;
    line-height: $unnnic-font-size-body-lg + $unnnic-line-height-md;
  }
}
</style>
