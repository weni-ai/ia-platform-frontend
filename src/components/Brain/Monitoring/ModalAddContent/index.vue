<template>
  <UnnnicModalDialog
    :modelValue="modelValue"
    size="lg"
    showCloseIcon
    :title="$t('router.monitoring.improve_response.add_new_content')"
    :secondaryButtonProps="{
      text: $t('cancel'),
    }"
    :primaryButtonProps="{
      text: $t('finish'),
    }"
    @update:model-value="$emit('update:modelValue', $event)"
    @secondary-button-click="$emit('update:modelValue', false)"
  >
    <UnnnicTab
      :tabs="contentTabs.map((tab) => tab.page)"
      :activeTab="activeTab"
      @change="onTabChange"
    >
      <template
        v-for="tab in contentTabs"
        :key="tab.page"
        #[`tab-head-${tab.page}`]
      >
        {{ $t(`content_bases.tabs.${tab.title}`) }}
      </template>
    </UnnnicTab>

    <section class="modal-add-content__content">
      <AddFilesContent v-show="activeTab === 'files'" />

      <AddSitesContent v-show="activeTab === 'sites'" />

      <textarea
        v-show="activeTab === 'text'"
        class="content__text"
        :placeholder="$t('content_bases.write_content_placeholder')"
        cols="30"
        rows="10"
        @input="$emit('update:modelValue', $event.target.value)"
      />
    </section>
  </UnnnicModalDialog>
</template>

<script setup>
import { ref } from 'vue';

import AddFilesContent from './AddFilesContent.vue';
import AddSitesContent from './AddSitesContent.vue';

const props = defineProps({
  modelValue: {
    type: Boolean,
    default: false,
  },
});

const emit = defineEmits(['close', 'update:modelValue']);

const contentTabs = ref([
  { title: 'files', page: 'files' },
  { title: 'sites', page: 'sites' },
  { title: 'text', page: 'text' },
]);
const activeTab = ref(contentTabs.value[0].page);
const onTabChange = (newTab) => {
  activeTab.value = newTab;
};
</script>

<style scoped lang="scss">
.modal-add-content {
  &__content {
    border-radius: $unnnic-border-radius-sm;
    border: $unnnic-border-width-thinner solid $unnnic-color-neutral-cleanest;

    padding: $unnnic-spacing-sm;

    min-height: 30vh;

    display: grid;
    align-items: center;

    .content__text {
      outline: none;
      resize: none;
      border: none;

      width: 100%;
      height: 100%;

      color: $unnnic-color-neutral-dark;
      font-family: $unnnic-font-family-secondary;
      font-size: $unnnic-font-size-body-gt;
      line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
      font-weight: $unnnic-font-weight-regular;

      &::placeholder {
        color: $unnnic-color-neutral-cloudy;
        font-family: $unnnic-font-family-secondary;
        font-size: $unnnic-font-size-body-gt;
        line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
        font-weight: $unnnic-font-weight-regular;
      }
    }
  }
}
</style>
