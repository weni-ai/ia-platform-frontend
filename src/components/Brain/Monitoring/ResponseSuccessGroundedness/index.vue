<template>
  <section
    class="response-success-groundedness"
    data-testid="response-success-groundedness"
  >
    <TextSentences
      :fragments="groundednessFragments"
      :getReliabilityLevel="getReliabilityLevel"
    />
    <GroundednessSources
      :fragments="groundednessFragments"
      :getReliabilityLevel="getReliabilityLevel"
    />
  </section>
</template>

<script setup>
import { computed } from 'vue';

import TextSentences from './TextSentences.vue';
import GroundednessSources from './GroundednessSources.vue';

const props = defineProps({
  inspectionData: {
    type: Object,
    required: true,
  },
});

const groundednessFragments = computed(() => {
  const data = props.inspectionData.groundedness.map((fragment) => ({
    ...fragment,
  }));

  if (data.length > 0) {
    const firstFragment = data[0];
    const lastFragment = data.at(-1);

    firstFragment.sentence = `“${firstFragment.sentence}`;
    lastFragment.sentence = `${lastFragment.sentence}”`;
  }

  return data;
});

function getReliabilityLevel(item) {
  const { score } = item;

  if (score === 0) return 'unreliable';
  if (score <= 33) return 'slightly-reliable';
  if (score <= 66) return 'moderately-reliable';
  if (score <= 100) return 'highly-reliable';
  return 'unreliable';
}
</script>

<style scoped lang="scss">
.response-success-groundedness {
  display: grid;
  gap: $unnnic-spacing-sm;
}
</style>
