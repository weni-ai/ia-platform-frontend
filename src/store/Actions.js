import nexusaiAPI from '@/api/nexusaiAPI.js';
import { defineStore } from 'pinia';
import { computed, reactive } from 'vue';
import globalStore from '.';

export const useActionsStore = defineStore('actions', () => {
  const connectProjectUuid = computed(
    () => globalStore.state.Auth.connectProjectUuid,
  );

  const actions = reactive({
    status: null,
    data: [],
  });

  async function loadActions() {
    if (actions.status !== null) {
      return;
    }

    try {
      actions.status = 'loading';

      const data = await nexusaiAPI.router.actions.list({
        projectUuid: connectProjectUuid.value,
      });

      actions.data = data;

      actions.status = 'complete';
    } catch (error) {
      actions.status = 'error';
    }
  }

  async function addAction({ name, prompt, flowUuid, type }) {
    const action = await nexusaiAPI.router.actions.create({
      projectUuid: connectProjectUuid.value,
      name,
      prompt,
      flowUuid,
      type,
    });

    actions.data.push(action);

    return action;
  }

  async function editAction({ uuid, name, prompt }) {
    const editedAction = await nexusaiAPI.router.actions.edit({
      projectUuid: connectProjectUuid.value,
      actionUuid: uuid,
      name,
      prompt,
    });

    const item = actions.data.find(
      (action) => action.uuid === editedAction.uuid,
    );

    item.name = editedAction.name;
    item.prompt = editedAction.prompt;

    return editedAction;
  }

  async function removeAction({ uuid }) {
    await nexusaiAPI.router.actions.delete({
      projectUuid: connectProjectUuid.value,
      actionUuid: uuid,
    });

    actions.data = actions.data.filter((action) => action.uuid !== uuid);
  }

  const types = reactive({
    status: null,
    data: [],
  });

  const typesAvailable = computed(() => {
    return types.data.filter(
      (type) =>
        !actions.data.some(
          (action) =>
            action.name === type.name && action.prompt === type.prompt,
        ),
    );
  });

  async function loadTypes() {
    if (types.status !== null) {
      return;
    }

    try {
      types.status = 'loading';

      const data = await nexusaiAPI.router.actions.types.list({
        projectUuid: connectProjectUuid.value,
      });

      types.data = data;

      types.status = 'complete';
    } catch {
      types.status = 'error';
    }
  }

  return {
    actions,
    load: loadActions,
    add: addAction,
    edit: editAction,
    remove: removeAction,

    types,
    typesAvailable,
    loadTypes,
  };
});
