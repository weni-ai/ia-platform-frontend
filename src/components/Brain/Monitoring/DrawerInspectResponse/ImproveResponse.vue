<template>
  <section class="improve-response">
    <header
      v-if="type === 'failed'"
      class="improve-response__header-failed"
      data-testid="header-failed"
    >
      <UnnnicIcon
        icon="cancel"
        scheme="aux-red-500"
        size="sm"
      />
      <UnnnicIntelligenceText
        color="aux-red-500"
        family="secondary"
        size="body-gt"
        tag="p"
      >
        {{ $t('router.monitoring.improve_response.agent_unable_respond') }}
      </UnnnicIntelligenceText>
    </header>
    <section
      :class="[
        'improve-response__actions',
        { 'improve-response__actions--with-header': type === 'failed' },
      ]"
    >
      <UnnnicIntelligenceText
        color="neutral-cloudy"
        family="secondary"
        size="body-gt"
        tag="p"
      >
        {{ $t('router.monitoring.improve_response.title') }}
      </UnnnicIntelligenceText>

      <section class="actions__buttons">
        <UnnnicButton
          v-if="actionToEdit"
          data-testid="edit-action-button"
          size="small"
          :text="
            $t('router.monitoring.improve_response.edit_action', {
              name: actionToEdit.name,
            })
          "
          type="secondary"
          @click="openEditModalAction"
        />
        <UnnnicButton
          v-else
          data-testid="add-content-button"
          size="small"
          :text="$t('router.monitoring.improve_response.add_new_content')"
          type="secondary"
          @click="isModalAddContentOpen = true"
        />
        <UnnnicButton
          data-testid="add-action-button"
          size="small"
          :text="$t('router.monitoring.improve_response.add_new_action')"
          type="secondary"
          @click="isModalAddActionOpen = true"
        />
      </section>
    </section>

    <ModalAddContent
      v-if="isModalAddContentOpen"
      v-model="isModalAddContentOpen"
      data-testid="modal-add-content"
    />

    <ModalActions
      v-if="isModalAddActionOpen"
      v-model="isModalAddActionOpen"
      data-testid="modal-add-action"
      actionGroup="custom"
      :actionToEditUuid="shouldEditAction ? actionToEdit?.uuid : undefined"
      @previous-step="isModalAddActionOpen = false"
    />
  </section>
</template>

<script setup>
import ModalAddContent from '../ModalAddContent/index.vue';

import { computed, onMounted, ref, watch } from 'vue';
import { useMonitoringStore } from '@/store/Monitoring';
import { useActionsStore } from '@/store/Actions';

import ModalActions from '@/components/actions/ModalActions.vue';

const props = defineProps({
  type: {
    type: String,
    required: true,
    validator: (value) => ['action', 'failed', 'success'].includes(value),
  },
});

const monitoringStore = useMonitoringStore();
const actionsStore = useActionsStore();

const actionToEdit = computed(
  () => monitoringStore.messages.inspectedAnswer?.action,
);

const isModalAddContentOpen = ref(false);
const isModalAddActionOpen = ref(false);
const shouldEditAction = ref(false);

function openEditModalAction() {
  isModalAddActionOpen.value = true;
  shouldEditAction.value = true;
}

watch(isModalAddActionOpen, (newIsModalAddActionOpen) => {
  if (!newIsModalAddActionOpen) {
    shouldEditAction.value = false;
  }
});

onMounted(() => {
  if (actionsStore.actions.status !== 'complete') {
    actionsStore.load();
  }
});
</script>

<style scoped lang="scss">
.improve-response {
  &__header-failed {
    border-radius: $unnnic-border-radius-sm $unnnic-border-radius-sm 0 0;

    padding: $unnnic-spacing-sm;

    background-color: $unnnic-color-aux-red-100;

    display: flex;
    gap: $unnnic-spacing-xs;
    align-items: center;
  }

  &__actions {
    border: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;
    border-radius: $unnnic-border-radius-sm;
    padding: $unnnic-spacing-sm;

    display: grid;
    gap: $unnnic-spacing-xs;

    &--with-header {
      border-radius: 0 0 $unnnic-border-radius-sm $unnnic-border-radius-sm;
      border-top: none;
    }

    .actions__buttons {
      display: grid;
      gap: $unnnic-spacing-nano;
    }
  }
}
</style>
