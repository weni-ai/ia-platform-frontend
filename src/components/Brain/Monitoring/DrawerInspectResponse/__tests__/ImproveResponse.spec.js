import { mount } from '@vue/test-utils';
import { describe, it, expect, vi, beforeEach } from 'vitest';
import { useMonitoringStore } from '@/store/Monitoring';
import { useActionsStore } from '@/store/Actions';
import i18n from '@/utils/plugins/i18n';

import ImproveResponse from '../ImproveResponse.vue';

vi.mock('@/store/Monitoring', () => ({
  useMonitoringStore: vi.fn(),
}));

vi.mock('@/store/Actions', () => ({
  useActionsStore: vi.fn(),
}));

const createWrapper = (props = {}) => {
  const mockMonitoringStore = {
    messages: {
      inspectedAnswer: {
        action: props.actionToEdit || null,
      },
    },
  };

  const mockActionsStore = {
    actions: {
      status: null,
    },
    load: vi.fn(),
  };

  useMonitoringStore.mockReturnValue(mockMonitoringStore);
  useActionsStore.mockReturnValue(mockActionsStore);

  return mount(ImproveResponse, {
    props: {
      type: props.type || 'success',
    },
    global: {
      stubs: ['UnnnicButton', 'UnnnicIcon', 'ModalAddContent', 'ModalActions'],
    },
  });
};

describe('ImproveResponse.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = createWrapper();
  });

  const headerFailed = () => wrapper.find('[data-testid="header-failed"]');
  const addContentButton = () =>
    wrapper.findComponent('[data-testid="add-content-button"]');
  const addActionButton = () =>
    wrapper.findComponent('[data-testid="add-action-button"]');
  const editActionButton = () =>
    wrapper.findComponent('[data-testid="edit-action-button"]');
  const modalAddAction = () =>
    wrapper.findComponent('[data-testid="modal-add-action"]');

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('calls actionsStore.load if actions.status is not complete on mounted', () => {
    const mockActionsStore = useActionsStore();

    wrapper = createWrapper();
    wrapper.vm.actionsStore.actions.status = null;
    expect(mockActionsStore.load).toHaveBeenCalled();

    mockActionsStore.load.mockReset();

    wrapper = createWrapper();
    wrapper.vm.actionsStore.actions.status = 'complete';

    expect(mockActionsStore.load).not.toHaveBeenCalled();
  });

  it('renders the failed header when type is "failed"', () => {
    wrapper = createWrapper({ type: 'failed' });

    expect(headerFailed().exists()).toBe(true);
    expect(headerFailed().text()).toContain(
      i18n.global.t('router.monitoring.improve_response.agent_unable_respond'),
    );
  });

  it('does not render the failed header when type is not "failed"', () => {
    wrapper = createWrapper({ type: 'success' });

    expect(headerFailed().exists()).toBe(false);
  });

  it('renders the buttons with correct text', () => {
    wrapper = createWrapper({ actionToEdit: { name: 'TestEditName' } });

    const editButton = wrapper.findComponent(
      '[data-testid="edit-action-button"]',
    );

    expect(editButton.exists()).toBe(true);
    expect(editButton.props('text')).toContain('TestEditName');
    expect(addContentButton().exists()).toBe(false);
    expect(addActionButton().exists()).toBe(true);
  });

  it('renders the add content button when there is no action to edit', () => {
    expect(addContentButton().exists()).toBe(true);
    expect(addContentButton().props('text')).toContain(
      i18n.global.t('router.monitoring.improve_response.add_new_content'),
    );
  });

  it('opens the modal when add content button is clicked', async () => {
    const modalAddContent = () =>
      wrapper.findComponent('[data-testid="modal-add-content"]');

    expect(modalAddContent().exists()).toBe(false);

    await addContentButton().trigger('click');

    expect(modalAddContent().exists()).toBe(true);
  });

  it('opens the modal when add action button is clicked', async () => {
    expect(modalAddAction().exists()).toBe(false);

    await addActionButton().trigger('click');

    expect(modalAddAction().exists()).toBe(true);
  });

  it('close the modal when @previous-step at ModalActions is emitted', async () => {
    await addActionButton().trigger('click');
    expect(wrapper.vm.isModalAddActionOpen).toBe(true);

    await modalAddAction().vm.$emit('previous-step');

    expect(wrapper.vm.isModalAddActionOpen).toBe(false);
  });

  it('should pass actionToEditUuid to ModalActions when shouldEditAction is true', async () => {
    wrapper = createWrapper({
      actionToEdit: { name: 'TestEditName', uuid: '123' },
    });

    await editActionButton().trigger('click');
    await wrapper.vm.$nextTick();

    expect(modalAddAction().exists()).toBe(true);
    expect(modalAddAction().props('actionToEditUuid')).toBe(
      wrapper.vm.actionToEdit?.uuid,
    );
  });

  it('should pass undefined to actionToEditUuid when shouldEditAction is false', async () => {
    wrapper = createWrapper({
      actionToEdit: { name: 'TestEditName', uuid: '123' },
    });

    await addActionButton().trigger('click');
    await wrapper.vm.$nextTick();

    expect(modalAddAction().exists()).toBe(true);
    expect(modalAddAction().props('actionToEditUuid')).toBe('');
  });

  it('should reset shouldEditAction when isModalAddActionOpen is closed', async () => {
    wrapper.vm.isModalAddActionOpen = true;
    wrapper.vm.shouldEditAction = true;
    await wrapper.vm.$nextTick();

    wrapper.vm.isModalAddActionOpen = false;
    await wrapper.vm.$nextTick();

    expect(wrapper.vm.shouldEditAction).toBe(false);
  });
});
