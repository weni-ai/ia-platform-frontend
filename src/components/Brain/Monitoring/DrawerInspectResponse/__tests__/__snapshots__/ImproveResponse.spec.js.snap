// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ImproveResponse.vue > matches the snapshot 1`] = `
"<section data-v-a6b14f5e="" class="improve-response">
  <!--v-if-->
  <section data-v-a6b14f5e="" class="improve-response__actions">
    <p data-v-ec8da4ad="" data-v-a6b14f5e="" class="unnnic-text unnnic-text--size-body-gt unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-cloudy">Improve the response</p>
    <section data-v-a6b14f5e="" class="actions__buttons">
      <unnnic-button-stub data-v-a6b14f5e="" size="small" text="Add new content" type="secondary" float="false" iconleft="" iconright="" iconcenter="" next="false" disabled="false" loading="false" data-testid="add-content-button"></unnnic-button-stub>
      <unnnic-button-stub data-v-a6b14f5e="" size="small" text="Add new action" type="secondary" float="false" iconleft="" iconright="" iconcenter="" next="false" disabled="false" loading="false" data-testid="add-action-button"></unnnic-button-stub>
    </section>
  </section>
  <!--v-if-->
</section>"
`;
